--- a/src/network/lookup_serv.c	2020-02-21 01:37:02.000000000 +0100
+++ b/src/network/lookup_serv2.c	2020-07-21 02:56:43.707608276 +0200
@@ -75,7 +75,16 @@
 	case ENOENT:
 	case ENOTDIR:
 	case EACCES:
-		return EAI_SERVICE;
+		f = __fopen_rb_ca("./services", &_f, _buf, sizeof _buf);
+		if (!f) switch (errno) {
+		case ENOENT:
+		case ENOTDIR:
+		case EACCES:
+			return EAI_SERVICE;
+		default:
+			return EAI_SYSTEM;
+		}
+		break;
 	default:
 		return EAI_SYSTEM;
 	}
