--- a/src/network/lookup_name.c	2020-02-21 01:37:02.000000000 +0100
+++ b/src/network/lookup_name.c	2020-07-21 02:54:43.245831485 +0200
@@ -57,7 +57,15 @@
 	case ENOENT:
 	case ENOTDIR:
 	case EACCES:
-		return 0;
+		f = __fopen_rb_ca("./hosts", &_f, _buf, sizeof _buf);
+		if (!f) switch (errno) {
+		case ENOENT:
+		case ENOTDIR:
+		case EACCES:
+			return 0;
+		default:
+			return EAI_SYSTEM;
+		}
 	default:
 		return EAI_SYSTEM;
 	}
